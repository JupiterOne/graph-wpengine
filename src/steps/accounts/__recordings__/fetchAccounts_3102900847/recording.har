{
  "log": {
    "_recordingName": "fetchAccounts",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "4af80a67b8b7715f75fc69566d0c1fb9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.wpengineapi.com/v1/user"
        },
        "response": {
          "bodySize": 119,
          "content": {
            "mimeType": "application/json",
            "size": 119,
            "text": "{\"id\":\"3937869a-09a0-4198-a1c7-d70c451c24fc\",\"email\":\"stefan@creativice.com\",\"first_name\":\"Stefan\",\"last_name\":\"Virag\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "119"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:28:42 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "0e1d179c-e95f-4fd1-bb5c-f2f5a2ba2bc5"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_n0pFhKIAMF53g="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2ecea-4df20f310d0da6a825d1b252"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 a0549c0c4be19f37804ee07fca51b10d.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "PEofLMwLC-krC5A1NIls-yxf1KfnUzSnwId0PjhLanbTNzmp2q9-sg=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:28:42.194Z",
        "time": 669,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 669
        }
      },
      {
        "_id": "b5a41916ddb8fd6f9a367486771ee186",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/accounts?limit=10&offset=0"
        },
        "response": {
          "bodySize": 112,
          "content": {
            "mimeType": "application/json",
            "size": 112,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\",\"name\":\"j1inc\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "112"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:28:43 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "3578bc27-3838-4460-a26c-9ec3b6bcffab"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_n00HBeoAMFU9A="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2eceb-0fd0c4e4414a5daa040a0110"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 f6773e0e5d339061619a1aeb9d029153.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "podix2UvqsAW69pR9ut6AkR8v35SP3Ya7fWL5NdbGHDivqdXh5w2gg=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:28:42.913Z",
        "time": 1067,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1067
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
