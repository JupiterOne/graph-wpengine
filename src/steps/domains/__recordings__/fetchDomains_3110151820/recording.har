{
  "log": {
    "_recordingName": "fetchDomains",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "7c0782aad0b9678323207b7733ae2209",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/installs?limit=10&offset=0"
        },
        "response": {
          "bodySize": 398,
          "content": {
            "mimeType": "application/json",
            "size": 398,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"30c47931-a91d-468a-a5dc-b9711f06c60b\",\"name\":\"jupiterone\",\"cname\":\"jupiterone.wpengine.com\",\"stable_ips\":null,\"status\":\"active\",\"account\":{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\"},\"php_version\":\"7.3\",\"site\":{\"id\":\"4d91f2dd-9743-421e-9bcf-20dcf0a195b2\"},\"environment\":\"production\",\"primary_domain\":\"jupiterone.com\",\"is_multisite\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "398"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:52:23 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "b7b1983b-d336-47c9-9d42-585f688af94a"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_rSmEgpoAMFTvg="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2f276-0778f73755cc5e831f39c56a"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 c218d36bf9cc1c36cb60fc8924dc8cc1.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "Nt00erYZkmWelAHTLRemetEmc3AU8_m76RQNpHdIXxT3_WbERmtQ9Q=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:52:22.841Z",
        "time": 750,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 750
        }
      },
      {
        "_id": "80fb0b7911edbf28dc48cf90a34041f6",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 398,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1//installs/30c47931-a91d-468a-a5dc-b9711f06c60b/domains?limit=10&offset=0"
        },
        "response": {
          "bodySize": 804,
          "content": {
            "mimeType": "application/json",
            "size": 804,
            "text": "{\"previous\":null,\"next\":null,\"count\":5,\"results\":[{\"name\":\"www.jupiterone.io\",\"duplicate\":false,\"id\":\"6eac0494-51d2-49be-8da1-ea6649bb50f8\",\"primary\":false,\"redirect_to\":{\"id\":\"c2663ce5-e1bf-42d3-8b78-89b032f1dd03\",\"name\":\"jupiterone.com\"}},{\"name\":\"jupiterone.io\",\"duplicate\":false,\"id\":\"b83dbe62-1698-4834-99ea-3bf38e12866b\",\"primary\":false,\"redirect_to\":{\"id\":\"c2663ce5-e1bf-42d3-8b78-89b032f1dd03\",\"name\":\"jupiterone.com\"}},{\"name\":\"www.jupiterone.com\",\"duplicate\":false,\"id\":\"0de8bdf6-2637-431c-9a2e-eb66a1f107e4\",\"primary\":false,\"redirect_to\":{}},{\"name\":\"jupiterone.com\",\"duplicate\":false,\"id\":\"c2663ce5-e1bf-42d3-8b78-89b032f1dd03\",\"primary\":true,\"redirect_to\":{}},{\"name\":\"jupiterone.wpengine.com\",\"duplicate\":false,\"id\":\"d1806903-21cd-4493-9dba-17968f347ffe\",\"primary\":false,\"redirect_to\":{}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "804"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:52:24 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "88ee4278-7126-49f8-88d1-85ec7751a475"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_rSzH9SIAMFbew="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2f278-2fc13f37529911bf37e226f7"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 183f775990c096ea769d4097f23cbad3.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "6NVbpdK8Dz0zQXn_TXZDKRfMavqa3PueOcjb9XEujPSFL5A7E1-GSA=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:52:23.662Z",
        "time": 1064,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1064
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
