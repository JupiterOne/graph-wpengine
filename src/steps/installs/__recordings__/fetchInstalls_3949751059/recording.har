{
  "log": {
    "_recordingName": "fetchInstalls",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "b5a41916ddb8fd6f9a367486771ee186",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/accounts?limit=10&offset=0"
        },
        "response": {
          "bodySize": 112,
          "content": {
            "mimeType": "application/json",
            "size": 112,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\",\"name\":\"j1inc\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "112"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 13:40:29 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "31f5416a-e5b7-4d7f-978f-0618baf200a2"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_7IIGUBIAMFiLA="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e30bcd-2a6c0c2409511c3c43ad8265"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 183f775990c096ea769d4097f23cbad3.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "qvmjxZt8afU0Kffes1hmnTFMe6WC2Mdx30bs3D76VBChssBcaZZKtw=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T13:40:28.626Z",
        "time": 1288,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1288
        }
      },
      {
        "_id": "97123c15bf93205301118bdc565d93ea",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 349,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/sites?limit=10&offset=0"
        },
        "response": {
          "bodySize": 354,
          "content": {
            "mimeType": "application/json",
            "size": 354,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"4d91f2dd-9743-421e-9bcf-20dcf0a195b2\",\"name\":\"jupiterone\",\"account\":{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\"},\"installs\":[{\"id\":\"30c47931-a91d-468a-a5dc-b9711f06c60b\",\"name\":\"jupiterone\",\"environment\":\"production\",\"cname\":\"jupiterone.wpengine.com\",\"php_version\":\"7.3\",\"is_multisite\":false}]}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "354"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 13:40:30 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "6fdfabab-3e48-4e4a-9541-cfa64606c2ff"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_7ITFcloAMFX-Q="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e30bce-3ee4aac6604edfec7f234c6a"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 fec87288bd626f32b2e1177a1ee70226.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "jVIrbaeZl2MXh5AZ0Stap0tCSAmXWkAU1BKhmYjScyxiRVMe3YT7ug=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T13:40:29.969Z",
        "time": 1080,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1080
        }
      },
      {
        "_id": "7c0782aad0b9678323207b7733ae2209",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/installs?limit=10&offset=0"
        },
        "response": {
          "bodySize": 398,
          "content": {
            "mimeType": "application/json",
            "size": 398,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"30c47931-a91d-468a-a5dc-b9711f06c60b\",\"name\":\"jupiterone\",\"cname\":\"jupiterone.wpengine.com\",\"stable_ips\":null,\"status\":\"active\",\"account\":{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\"},\"php_version\":\"7.3\",\"site\":{\"id\":\"4d91f2dd-9743-421e-9bcf-20dcf0a195b2\"},\"environment\":\"production\",\"primary_domain\":\"jupiterone.com\",\"is_multisite\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "398"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 13:40:31 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "f37dc5c1-5f7a-4db2-acd7-cf984efb1a52"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_7IaE81oAMF7QA="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e30bcf-45a225a27748cb9f094ee0d0"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 a0549c0c4be19f37804ee07fca51b10d.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "LNeDQjkZ2V8RONqUIuUYOAAi3crJYwgaICvaF_8I5NHD7rBDDOU5VA=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T13:40:31.070Z",
        "time": 666,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 666
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
