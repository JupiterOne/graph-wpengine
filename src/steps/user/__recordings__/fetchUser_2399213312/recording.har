{
  "log": {
    "_recordingName": "fetchUser",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "4af80a67b8b7715f75fc69566d0c1fb9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.wpengineapi.com/v1/user"
        },
        "response": {
          "bodySize": 119,
          "content": {
            "mimeType": "application/json",
            "size": 119,
            "text": "{\"id\":\"3937869a-09a0-4198-a1c7-d70c451c24fc\",\"email\":\"stefan@creativice.com\",\"first_name\":\"Stefan\",\"last_name\":\"Virag\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "119"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Fri, 18 Jun 2021 15:07:33 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "d204b93a-2d7f-461f-bd8d-671e34cf15a3"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "BIF8THXhIAMFclw="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60ccb6b4-0596b255286e49bb1a1f1cef"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 881b12332738e10f6e80298fbdcd7e8f.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "VIE50-C2"
            },
            {
              "name": "x-amz-cf-id",
              "value": "MIjnbkgnc_Dl5H_XKCCMwLtognKlLK9VgZ0cdTWTCtwVXWu_iUrQ6Q=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-06-18T15:07:32.577Z",
        "time": 773,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 773
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
