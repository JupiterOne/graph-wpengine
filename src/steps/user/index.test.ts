import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { fetchUser } from '.';
import { integrationConfig } from '../../../test/config';
import { setupWPEngineRecording } from '../../../test/recording';

describe('#fetchUser', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupWPEngineRecording({
      directory: __dirname,
      name: 'fetchUser',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUser(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const users = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('User'),
    );

    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'wp_engine_user' },
          name: { type: 'string' },
          username: { type: 'string' },
          email: { type: 'string' },
          firstName: { type: 'string' },
          lastName: { type: 'string' },
        },
      },
    });
  });
});
