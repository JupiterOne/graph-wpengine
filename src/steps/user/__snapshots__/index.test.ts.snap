// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchUser should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "wp_engine_user:3937869a-09a0-4198-a1c7-d70c451c24fc",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "email": "stefan@creativice.com",
            "first_name": "Stefan",
            "id": "3937869a-09a0-4198-a1c7-d70c451c24fc",
            "last_name": "Virag",
          },
        },
      ],
      "_type": "wp_engine_user",
      "createdOn": undefined,
      "displayName": "Stefan Virag",
      "email": "stefan@creativice.com",
      "firstName": "Stefan",
      "id": "3937869a-09a0-4198-a1c7-d70c451c24fc",
      "lastName": "Virag",
      "name": "Stefan Virag",
      "username": "stefan@creativice.com",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "wp_engine_user",
  ],
  "numCollectedEntities": 1,
  "numCollectedRelationships": 0,
}
`;
